core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackSyncService.java

```diff
@@ -206,24 +206,20 @@ private void deleteResourceIfNeeded(Stack stack, InstanceMetaData instance) {
     }
 
     private void handleSyncResult(Stack stack, Map<InstanceSyncState, Integer> instanceStateCounts, boolean stackStatusUpdateEnabled) {
-        Set<InstanceMetaData> instances = instanceMetaDataRepository.findNotTerminatedForStack(stack.getId());
         if (instanceStateCounts.get(InstanceSyncState.UNKNOWN) > 0) {
             eventService.fireCloudbreakEvent(stack.getId(), AVAILABLE.name(),
                     cloudbreakMessagesService.getMessage(Msg.STACK_SYNC_INSTANCE_STATUS_COULDNT_DETERMINE.code()));
         } else if (instanceStateCounts.get(InstanceSyncState.IN_PROGRESS) > 0) {
             eventService.fireCloudbreakEvent(stack.getId(), AVAILABLE.name(),
                     cloudbreakMessagesService.getMessage(Msg.STACK_SYNC_INSTANCE_OPERATION_IN_PROGRESS.code()));
-        } else if (instanceStateCounts.get(InstanceSyncState.RUNNING) > 0 && instanceStateCounts.get(InstanceSyncState.STOPPED) > 0) {
+        } else if ((instanceStateCounts.get(InstanceSyncState.RUNNING) >= 0 || (instanceStateCounts.get(InstanceSyncState.DELETED_ON_PROVIDER_SIDE) >= 0))
+                && instanceStateCounts.get(InstanceSyncState.STOPPED) > 0) {
             eventService.fireCloudbreakEvent(stack.getId(), STOPPED.name(),
                     cloudbreakMessagesService.getMessage(Msg.STACK_SYNC_INSTANCE_STOPPED_ON_PROVIDER.code()));
         } else if (instanceStateCounts.get(InstanceSyncState.RUNNING) > 0) {
             updateStackStatusIfEnabled(stack.getId(), DetailedStackStatus.AVAILABLE, SYNC_STATUS_REASON, stackStatusUpdateEnabled);
             eventService.fireCloudbreakEvent(stack.getId(), AVAILABLE.name(),
                     cloudbreakMessagesService.getMessage(Msg.STACK_SYNC_INSTANCE_STATE_SYNCED.code()));
-        } else if (instanceStateCounts.get(InstanceSyncState.STOPPED).equals(instances.size())) {
```

Comment:
what if every nodes are stopped?

Reply:
it will remain stopped. Did I miss something?

Reply:
Which if clause handles that use-case? Why was the original "else if" deleted?

Reply:
I thought it will be the same as I did, I can fix that